cli/src/cli_util.rs

```diff
@@ -855,6 +896,29 @@ pub struct WorkspaceCommandHelper {
     working_copy_shared_with_git: bool,
 }
 
+enum SnapshotWorkingCopyError {
+    Command(CommandError),
+    StaleWorkingCopy(CommandError),
+}
+
+impl SnapshotWorkingCopyError {
+    fn into_command_error(self) -> CommandError {
+        match self {
+            Self::Command(err) => err,
+            Self::StaleWorkingCopy(err) => err,
+        }
+    }
+}
+
+impl<E> From<E> for SnapshotWorkingCopyError
+where
+    E: Into<CommandError>,
+{
+    fn from(value: E) -> Self {
+        Self::Command(value.into())
+    }
```

Comment:
nit: I wouldn't add automatic conversion because the caller should be careful to not map stale wc error to `Command(_)`.

Reply:
Ok, I don't think it's terrible to insert `.map_err(to_other_err)?`, but I agree it would look a bit unusual. We can extract `StaleWorkingCopyError` that isn't convertible to `CommandError`, but I don't have a strong opinion about that. We can refactor it later.

(No action requested)

Reply:
I like it, thanks.